test/unit/query/builder.js

```diff
@@ -8149,4 +8149,59 @@ describe('QueryBuilder', function() {
       oracledb: 'select 0',
     });
   });
+
+  it('join with onVal andOnVal orOnVal', () => {
+    testsql(
+      qb()
+        .select({
+          id: 'p.ID',
+          status: 'p.post_status',
+          name: 'p.post_title',
+          // type: 'terms.name',
+          price: 'price.meta_value',
+          createdAt: 'p.post_date_gmt',
+          updatedAt: 'p.post_modified_gmt',
+        })
+        .from({ p: 'wp_posts' })
+        .leftJoin({ price: 'wp_postmeta' }, function() {
+          this.on('p.id', '=', 'price.post_id')
+            .onVal(function() {
+              this.onVal('price.meta_key', '_regular_price').andOnVal(
+                'price_meta_key',
+                '_regular_price'
+              );
+            })
+            .orOnVal(function() {
```

Comment:
Why does this need the extra closure here?

Reply:
It doesn't necessarily. It's just personal preference. I usually prefer keeping conditions within bounds of closures so that future additions don't accidentally end up in the wrong closure. 

Calling without function would also yield valid syntax.